pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -213,24 +213,31 @@ func (kc *KubeConfig) UseWorkspace(ctx context.Context, name string) (err error)
 		}
 
 		if strings.Contains(name, ":") && cluster.HasPrefix(tenancyv1alpha1.RootCluster) {
-			// absolute logical cluster under root:
-			parentClusterName, workspaceName := logicalcluster.New(name).Split()
-			ws, err := kc.clusterClient.Cluster(parentClusterName).TenancyV1beta1().Workspaces().Get(ctx, workspaceName, metav1.GetOptions{})
-			if err != nil {
+			// e.g. root:something
+			groups, err := kc.clusterClient.Cluster(cluster).Discovery().ServerGroups()
+			if err != nil && !apierrors.IsForbidden(err) {
 				return err
 			}
+			if apierrors.IsForbidden(err) || len(groups.Groups) == 0 {
+				return fmt.Errorf("access to workspace %s denied or workspace does not exist", name)
+			}
```

Comment:
doesn't it make sense to differentiate the two cases?

Reply:
we cannot. In either case you get 403 from the server.

Reply:
actually, partially we can. sec